### ISSUE TITLE
[SidePanel] reports findDOMNode deprecation warning in StrictMode

### ISSUE THREAD

**Issue by cjkarle** [carbon-design-system/ibm-cloud-cognitive#1004]
## What package(s) are you using?

- [x] Cloud & Cognitive
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

The SidePanel component is reporting a StrictMode deprecation warning about findDOMNode as per this trace:

```
Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of ResizeDetector which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://fb.me/react-strict-mode-find-node
    in div (created by SidePanel)
    in ResizeDetector (created by SidePanel)
    in SidePanel (at <ourCode>.js:220)
```

Although such a warning will not be displayed in production (according to the strict mode check docs linked in the trace), we wanted to log the issue and track that the suggested updates are addressed in a future version.

> What version of the @carbon/ibm-cloud-cognitive package are you using?

0.54.1

## Steps to reproduce the issue

1. Incorporate SidePanel into a page and run the site with strict-mode enabled
2. Open the SidePanel

**Comment by dcwarwick** [comment]
@cjkarle Thank you for raising this issue. The use of `findDOMNode` is documented in the documentation for `ReactResizeDetector`, and there are ways we can provide the information the resize detector needs which will enable it not to require use of this deprecated call. As it happens, I am right now refactoring the use of `ReactResizeDetector` in our components to use it more efficiently, so as part of that we will ensure that in particular it doesn't use `findDOMNode` any more.

Additionally, we should be checking for other strict mode warnings -- storybook added support for a strict mode flag in its version 6.1.0, and we recently upgraded our storybook past that point. I therefore propose two tasks to address this issue:
- [ ] Enable the strict mode flag in storybook, and verify that this produces the same warnings from SidePanel.
- [ ] Refactor the use of `ReactResizeDetector` in our components and ensure that as well as a more efficient hook-based usage pattern each component produces no strict-mode warnings.

**Comment by dcwarwick** [comment]
Issue resolved by #1009. All use of `ReactResizeDetector` now uses the hook pattern and should not require `findDOMNode`.
